StRoot/StFcsTrackMatchMaker/StFcsTrackMatchMaker.cxx

```diff
@@ -0,0 +1,250 @@
+    mXY[1]=new TH2F("XY_H","XY_H",50,-130,130,50,-110,110);
+    mXY[2]=new TH2F("XY","XY",50,-130,130,50,-110,110);
+  }
+  return kStOK;
+}
+
+int StFcsTrackMatchMaker::Finish(){
+   if(mEpdgeo) delete mEpdgeo; 
+  if(mFile){
+    LOG_INFO << "Closing "<<mFilename<<endm;
+    mFile->Write();
+    mFile->Close();
+  }
+  return kStOK;
+}
+
+int StFcsTrackMatchMaker::Make(){
+  StEvent* event = (StEvent*)GetInputDS("StEvent");
+  if(!event) {
+    LOG_ERROR << "StFcsTrackMatchMaker::Make did not find StEvent"<<endm; 
+    return kStErr;
+  }
+  mFcsColl = event->fcsCollection();
+  if(!mFcsColl) {
+    LOG_ERROR << "StFcsTrackMatchMaker::Make did not find StEvent->StFcsCollection"<<endm; 
+    return kStErr;
+  }
+  mFwdTrkColl = event->fwdTrackCollection();
+  if(!mFwdTrkColl) {
+    LOG_ERROR << "StFcsTrackMatchMaker::Make did not find StEvent->fwdTrackCollection"<<endm; 
+    return kStErr;
+  }
+    
+  int ntrk=mFwdTrkColl->numberOfTracks();
+  int ngoodtrk=0;
+  int nMatch[2]={0,0};
+  for(int itrk=0; itrk<ntrk; itrk++){
+    StFwdTrack* trk=mFwdTrkColl->tracks()[itrk];
+    if(trk->didFitConvergeFully()==false) continue;
+    ngoodtrk++;
+    const StThreeVectorD* proj[3];
+    proj[0] = trk->getProjection(0)->getXYZ(); //Ecal
+    proj[1] = trk->getProjection(1)->getXYZ(); //Hcal
+    proj[2] = trk->getProjection(2)->getXYZ(); //Pres=Epdw
+    if(Debug()){      
+      LOG_INFO << Form("Proj0 : %6.2f %6.2f %6.2f",proj[0]->x(),proj[0]->y(),proj[0]->z())<<endm;
+      LOG_INFO << Form("Proj1 : %6.2f %6.2f %6.2f",proj[1]->x(),proj[1]->y(),proj[1]->z())<<endm;
+      LOG_INFO << Form("Proj2 : %6.2f %6.2f %6.2f",proj[2]->x(),proj[2]->y(),proj[2]->z())<<endm;
+    }
+    
+    //North or south from track
+    int ns=0;
+    if(proj[0]->x()>0.0) ns=1;
+
+    //Look for a Ecal & Hcal match for a track
+    for(int ehp=0; ehp<2; ehp++){
+      int det = ehp*2 + ns;
+      int nclu  = mFcsColl->numberOfClusters(det);
+      for(int iclu=0; iclu<nclu; iclu++){
+	StFcsCluster* clu=mFcsColl->clusters(det)[iclu];
+	if(clu->energy() > mMinEnergy[ehp]){
+	  StThreeVectorD xyz=mFcsDb->getStarXYZfromColumnRow(det,clu->x(),clu->y());
+	  if(Debug()){
+	    double dx = xyz.x() - proj[ehp]->x();
+	    double dy = xyz.y() - proj[ehp]->y();
+	    double dr = sqrt(dx*dx + dy*dy);
+	    if(Debug()) LOG_INFO << Form("EHP=%1d dx = %6.2f - %6.2f  = %6.2f dy = %6.2f - %6.2f  = %6.2f dr=%6.2f",
+					 ehp,xyz.x(),proj[ehp]->x(),dx,xyz.y(),proj[ehp]->y(),dy,dr) << endm;
+	  };
+	  if(mFile){
+	    double dx = xyz.x() - proj[ehp]->x();
+	    double dy = xyz.y() - proj[ehp]->y();
+	    double dr = sqrt(dx*dx + dy*dy);	    
+	    if(fabs(dy)<mMaxDistance[ehp]) mHdx[ehp]->Fill(dx);
+	    if(fabs(dx)<mMaxDistance[ehp]) mHdy[ehp]->Fill(dy);
+	    mHdr[ehp]->Fill(dr);
+	  }
+	  double dx = xyz.x() - proj[ehp]->x();  if(dx > mMaxDistance[ehp]) continue;
+	  double dy = xyz.y() - proj[ehp]->y();  if(dy > mMaxDistance[ehp]) continue;
```

Comment:
Do we need these two lines? The third line "dr" cut will automatically satisfy these cuts. If so, do we need to calculate dx = TMath::Abs(dx) and dy = TMath::Abs(dy) before comparing to the max distance?